### PR TITLE
Remove (one) redundant parenthesization

### PR DESCRIPTION
**Pull request by shayne-fletcher-da**
One redundant parenthesization removed from the record processor.

### CODE REVIEW

File: compiler/damlc/daml-preprocessor/src/DA/Daml/Preprocessor/Records.hs [hunk 1]
```diff
@@ -117,16 +117,22 @@ onDecl o = descendBi onExp o : extraDecls o
...
         ctor (L _ ConDeclH98{con_args=RecCon (L _ fields),con_name=L _ name}) = concatMap (field name) fields
-        ctor _ = []
+        ctor _ = [] -- GADT ctors ignored.
 
         field name (L _ ConDeclField{cd_fld_type=L _ ty, ..}) = [(result, name, fld, ty) | L _ fld <- cd_fld_names]
         field _ _ = error "unknown field declaration in getFields"
 
-        result = noL $ HsParTy noE $ foldl (\x y -> noL $ HsAppTy noE x $ hsLTyVarBndrToType y) (noL $ HsTyVar noE GHC.NotPromoted tcdLName) $ hsq_explicit tcdTyVars
+        -- A value of this data declaration will have this type.
+        result = foldl (\x y -> noL $ HsAppTy noE x $ hsLTyVarBndrToType y)
```
Comment (cocreature) [inline compiler/damlc/daml-preprocessor/src/DA/Daml/Preprocessor/Records.hs]:
```suggestion
        result = foldl' (\x y -> noL $ HsAppTy noE x $ hsLTyVarBndrToType y)
```

File: compiler/damlc/daml-preprocessor/src/DA/Daml/Preprocessor/Records.hs [hunk 2]
```diff
@@ -117,16 +117,22 @@ onDecl o = descendBi onExp o : extraDecls o
...
+-- Process a Haskell data type definition. Walk its constructors. For
+-- each, harvest a tuple of the parent type, the ctor name, the field
+-- name and the field type.
+getFields :: TyClDecl GhcPs ->
+               [(LHsType GhcPs, IdP GhcPs, FieldOcc GhcPs, HsType GhcPs)]
 getFields DataDecl{tcdDataDefn=HsDataDefn{..}, ..} = concatMap ctor dd_cons
     where
         ctor (L _ ConDeclH98{con_args=RecCon (L _ fields),con_name=L _ name}) = concatMap (field name) fields
-        ctor _ = []
+        ctor _ = [] -- GADT ctors ignored.
```
Comment (hurryabit) [inline compiler/damlc/daml-preprocessor/src/DA/Daml/Preprocessor/Records.hs]: Leverage the compiler to "check" the comment:
```suggestion
        ctor (L _ ConDeclGADT{}) = []
```

Comment (shayne-fletcher-da) [reply compiler/damlc/daml-preprocessor/src/DA/Daml/Preprocessor/Records.hs]: ✔️

File: compiler/damlc/daml-preprocessor/src/DA/Daml/Preprocessor/Records.hs (hunk 1 above)
Comment (shayne-fletcher-da) [reply compiler/damlc/daml-preprocessor/src/DA/Daml/Preprocessor/Records.hs]: ✔️

**Pull request merged**